### PR TITLE
Add {allow,deny}-address options

### PR DESCRIPTION
**Pull request by andrew-stripe**
This adds `--allow-address` and `--deny-address` options that operate similarly to the range options but allow for specifying a particular port to allow/block. My immediate use for this is allowing the proxy to chain to another proxy running on localhost while preventing it from accessing other services on localhost.

### CODE REVIEW

File: pkg/smokescreen/config.go [hunk 1]
```diff
@@ -13,18 +13,24 @@ import (
...
 	"os"
+	"strconv"
 	"sync"
 	"time"
 
 	"github.com/DataDog/datadog-go/statsd"
 	log "github.com/sirupsen/logrus"
 )
 
+type RuleRange struct {
```
Comment (andrew-stripe) [inline pkg/smokescreen/config.go]: I think `RuleRange` is kind of a weak name. Open to ideas.

Comment (rlk-stripe) [reply pkg/smokescreen/config.go]: `RangeWithPort` maybe?  I'm pretty ok with `RuleRange` though.

**Comment by rlk-stripe** [comment]
LGTM

File: pkg/smokescreen/config.go (hunk 1 above)
Comment (andrew-stripe) [reply pkg/smokescreen/config.go]: I can leave as-is if you think it's OK. Sometimes I over-think the naming...

**Pull request merged**